src/core/components/checkbox/index.tsx

```diff
@@ -39,13 +45,14 @@ const CheckboxGroup = ({
 	id,
 	name,
 	label,
+	hideLabel,
 	supporting,
 	error,
 	cssOverrides,
 	children,
 	...props
 }: CheckboxGroupProps) => {
-	const legend = label ? <Legend text={label} supporting={supporting} /> : ""
+	const legend = label ? <Legend text={label} supporting={supporting} cssOverrides={hideLabel ? visuallyHidden : undefined} /> : ""
```

Comment:
Nice work discovering the `cssOverrides` prop. However I think in this case, I'd rather there be a `visuallyHidden: boolean` prop that you pass to the Legend component. This would be tidier, I think, and less prone to error in the future (`cssOverrides` is intended as an escape hatch)